`make` fails on working directories with spaces in them

The location I originally cloned minko into had a space:

`/media/3TB SATA SEAGATE/Documents/Code/repos`

When I attempted to make it, the following error happened.

    ~/repos/minko♠ make config=release config=linux64_release
    ==== Building minko-framework (linux64_release) ====
    Creating bin/linux64/release
    Creating obj/linux64/release
    jsoncpp.cpp
    /bin/sh: /media/3TB: No such file or directory
    Makefile:270: recipe for target 'obj/linux64/release/jsoncpp.o' failed
    make[1]: *** [obj/linux64/release/jsoncpp.o] Error 127
    Makefile:332: recipe for target 'minko-framework' failed
    make: *** [minko-framework] Error 2

A simple workaround is to use a location without a space, obviously, but I thought I'd make mention of it.

Sure!

    /media/3TB SATA SEAGATE/Documents/Code/repos♠ git clone https://github.com/aerys/minko.git minko-test
    Cloning into 'minko-test'...
    remote: Counting objects: 90112, done.
    remote: Compressing objects: 100% (10/10), done.
    remote: Total 90112 (delta 3), reused 0 (delta 0)
    Receiving objects: 100% (90112/90112), 172.28 MiB | 3.97 MiB/s, done.
    Resolving deltas: 100% (49632/49632), done.
    Checking connectivity... done.
    /media/3TB SATA SEAGATE/Documents/Code/repos♠ cd minko-test/
    /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test♠ ./tool/lin/script/solution_gmake_gcc.sh
    ./tool/lin/script/solution_gmake_gcc.sh: line 5: pushd: /media/3TB: No such file or directory
    Minko SDK home directory: /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test
    You must define the environment variable EMSCRIPTEN to be able to target HTML5.
    You must define the environment variable ANDROID_HOME to be able to target Android.
    Building configurations...
    Running action 'gmake'...
    Generating Makefile...
    Generating framework/Makefile...
    Generating tutorial/01-hello-cube/Makefile...
    Generating tutorial/02-handle-canvas-resizing/Makefile...
    Generating tutorial/03-rotating-the-camera-around-an-object-with-the-mouse/Makefile...
    Generating tutorial/04-moving-objects/Makefile...
    Generating tutorial/05-moving-objects-with-the-keyboard/Makefile...
    Generating tutorial/06-load-3d-files/Makefile...
    Generating tutorial/07-loading-scene-files/Makefile...
    Generating tutorial/08-my-first-script/Makefile...
    Generating tutorial/09-scripting-mouse-inputs/Makefile...
    Generating tutorial/10-working-with-the-basic-material/Makefile...
    Generating tutorial/11-working-with-the-phong-material/Makefile...
    Generating tutorial/12-working-with-normal-maps/Makefile...
    Generating tutorial/13-working-with-environment-maps/Makefile...
    Generating tutorial/14-working-with-specular-maps/Makefile...
    Generating tutorial/15-loading-and-using-textures/Makefile...
    Generating tutorial/16-loading-effects/Makefile...
    Generating tutorial/17-creating-a-custom-effect/Makefile...
    Generating tutorial/18-creating-custom-materials/Makefile...
    Generating tutorial/19-binding-the-model-to-world-transform/Makefile...
    Generating tutorial/20-binding-the-camera/Makefile...
    Generating tutorial/21-authoring-uber-shaders/Makefile...
    Generating tutorial/22-creating-a-simple-post-processing-effect/Makefile...
    Generating tutorial/23-using-external-glsl-code-in-effect-files/Makefile...
    Generating tutorial/24-working-with-custom-vertex-attributes/Makefile...
    Generating tutorial/25-working-with-ambient-lights/Makefile...
    Generating tutorial/26-working-with-directional-lights/Makefile...
    Generating tutorial/27-working-with-point-lights/Makefile...
    Generating tutorial/28-working-with-spot-lights/Makefile...
    Generating tutorial/29-hello-falling-cube/Makefile...
    Generating tutorial/30-applying-anti-aliasing-effect/Makefile...
    Generating plugin/lua/Makefile...
    Generating plugin/zlib/Makefile...
    Generating plugin/assimp/Makefile...
    Generating plugin/devil/Makefile...
    Generating plugin/bullet/Makefile...
    Generating plugin/fx/Makefile...
    Generating plugin/html-overlay/Makefile...
    Generating plugin/http-loader/Makefile...
    Generating plugin/http-worker/Makefile...
    Generating plugin/jpeg/Makefile...
    Generating plugin/leap/Makefile...
    Generating plugin/oculus/Makefile...
    Generating plugin/particles/Makefile...
    Generating plugin/png/Makefile...
    Generating plugin/sdl/Makefile...
    Generating plugin/serializer/Makefile...
    Generating example/assimp/Makefile...
    Generating example/clone/Makefile...
    Generating example/cube/Makefile...
    Generating example/devil/Makefile...
    Generating example/effect-config/Makefile...
    Generating example/flares/Makefile...
    Generating example/fog/Makefile...
    Generating example/frustum/Makefile...
    Generating example/hologram/Makefile...
    Generating example/html-overlay/Makefile...
    Generating example/http/Makefile...
    Generating example/jobs/Makefile...
    Generating example/joystick/Makefile...
    Generating example/keyboard/Makefile...
    Generating example/leap-motion/Makefile...
    Generating example/light/Makefile...
    Generating example/line-geometry/Makefile...
    Generating example/lua-scripts/Makefile...
    Generating example/multi-surfaces/Makefile...
    Generating example/oculus/Makefile...
    Generating example/particles/Makefile...
    Generating example/physics/Makefile...
    Generating example/picking/Makefile...
    Generating example/raycasting/Makefile...
    Generating example/serializer/Makefile...
    Generating example/sky-box/Makefile...
    Generating example/stencil/Makefile...
    Generating example/visibility/Makefile...
    Generating example/water/Makefile...
    Done.
    ./tool/lin/script/solution_gmake_gcc.sh: line 7: popd: directory stack empty
    /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test♠ make config=release config=linux64_release
    ==== Building minko-framework (linux64_release) ====
    Creating bin/linux64/release
    Creating obj/linux64/release
    jsoncpp.cpp
    /bin/sh: /media/3TB: No such file or directory
    Makefile:270: recipe for target 'obj/linux64/release/jsoncpp.o' failed
    make[1]: *** [obj/linux64/release/jsoncpp.o] Error 127
    Makefile:332: recipe for target 'minko-framework' failed
    make: *** [minko-framework] Error 2
    /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test♠ make config=linux64_release verbose=1
    ==== Building minko-framework (linux64_release) ====
    jsoncpp.cpp
    /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test/tool/lin/script/g++.sh g++    -MMD -MP -DNDEBUG -DJSON_IS_AMALGAMATION -Iinclude -Isrc -Ilib/jsoncpp/src  -m64 -O2 -std=c++11  -o "obj/linux64/release/jsoncpp.o" -MF obj/linux64/release/jsoncpp.d -c "lib/jsoncpp/src/jsoncpp.cpp"
    /bin/sh: /media/3TB: No such file or directory
    Makefile:270: recipe for target 'obj/linux64/release/jsoncpp.o' failed
    make[1]: *** [obj/linux64/release/jsoncpp.o] Error 127
    Makefile:332: recipe for target 'minko-framework' failed
    make: *** [minko-framework] Error 2

I called both. Please see the last nine lines.

    /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test♠ make config=linux64_release verbose=1
    ==== Building minko-framework (linux64_release) ====
    jsoncpp.cpp
    /media/3TB SATA SEAGATE/Documents/Code/repos/minko-test/tool/lin/script/g++.sh g++    -MMD -MP -DNDEBUG -DJSON_IS_AMALGAMATION -Iinclude -Isrc -Ilib/jsoncpp/src  -m64 -O2 -std=c++11  -o "obj/linux64/release/jsoncpp.o" -MF obj/linux64/release/jsoncpp.d -c "lib/jsoncpp/src/jsoncpp.cpp"
    /bin/sh: /media/3TB: No such file or directory
    Makefile:270: recipe for target 'obj/linux64/release/jsoncpp.o' failed
    make[1]: *** [obj/linux64/release/jsoncpp.o] Error 127
    Makefile:332: recipe for target 'minko-framework' failed
    make: *** [minko-framework] Error 2

--- a/module/gcc/gcc.lua
+++ b/module/gcc/gcc.lua
@@ -16,7 +16,7 @@ table.inject(premake.tools.gcc, 'cxxflags.system.macosx', {
 
 table.inject(premake.tools.gcc, 'tools.linux', {
 	cc = 'gcc',
-	cxx = MINKO_HOME .. '/tool/lin/script/g++.sh g++',
+	cxx = '\''..MINKO_HOME .. '/tool/lin/script/g++.sh\' g++',
 })
 
 table.inject(premake.tools.gcc, 'tools.macosx', {

